python script to send reminder mail for activation link

kirti3192 opened the pull request:
This script will check for users who registered min 7 days ago and still their activation status is zero. 
Script will send reminder mail to all these listed users and store logs.

theskumar commented on cron/activation-reminder-mail.py:
```diff
@@ -0,0 +1,65 @@
+
+Thank you for registering on Spoken Tutorial website.
+This is a gentle reminder from Spoken Tutorial Team to activate your account. 
+Kindly activate your account by clicking on the activation link which has been sent to you at the time of registration.
```
Let's put the activation link here only, so user do not have search his/her email again.

theskumar replied:
Nice to have: a link to opt-out of reminder emails.

theskumar commented:
Use management command instead of pure python.

theskumar commented:
- Don't send email to user after 6months.

kirti3192 commented:
@theskumar : I have done all the changes as per your suggestions. Kindly check my commit and review. Please check for syntax and logical changes.
Remaining part:
1. To add NO_REPLY_MAIL in config file (online)
2. To create reminder_mail_success_log.txt file on server (where to locate this)
3. Creating doc for file

theskumar commented on cms/management/commands/activation_reminder_mail.py:
```diff
@@ -0,0 +1,79 @@
+from django.db import transaction as tx
+from django.contrib.auth.models import User
+from cms.models import Profile
+from config import *
```
Instead of `from config import *`

You should be using 
```
from django.conf import settings

# then use it like 
settings.NO_REPLY_EMAIL
```

Also, check what is the value of  `DEFAULT_EMAIL_FROM` [1], you can use the same instead of introducing a new variable. 

[1] https://docs.djangoproject.com/en/1.10/ref/settings/#std:setting-DEFAULT_FROM_EMAIL

kirti3192 replied:
There is no default email. We have something called administrator email and it is not no-reply

kirti3192 replied:
will create NO_REPLY_EMAIL in settings.py

theskumar commented on cms/management/commands/activation_reminder_mail.py:
```diff
@@ -0,0 +1,79 @@
+                user.save()
+                print str(user.id),',', str(user.email),',', str(1)
+                if sent%100 == 0:
+                    time.sleep(10)
```
This is something that should be handled higher the level, i.e. email backend or SMTP, right now we do have a delay enabled in the SMTP.

kirti3192 replied:
ok I'll remove it from here

theskumar commented on cms/management/commands/activation_reminder_mail.py:
```diff
@@ -0,0 +1,79 @@
+                sent += 1
+                user.reminder_last_sent = today
+                user.save()
+                print str(user.id),',', str(user.email),',', str(1)
```
I think you can get-rid of `str(1)`?

theskumar replied:
It would be nice if you use `print()` as function, since you are using `from __future__ import print_function`

kirti3192 replied:
str(1) or (0) is just to rectify sent or not sent

theskumar commented on cms/management/commands/activation_reminder_mail.py:
```diff
@@ -0,0 +1,79 @@
+                subject, message, NO_REPLY_MAIL,
+                to = to,
+                headers = {
+                 "Content-type" : "text/html"
```
When you say it's a html email, I think you have missed the html markups in the email, else you should be using plain text emails.

theskumar commented:
It seems you have not added `NO_REPLY_EMAIL` to the setting files. Can you add that added and the work with @manivel23  to deploy and test it? 

You can merge this PR yourself when you want! 👍

kirti3192 commented:
@theskumar : added no-reply variable in settings file.
As per @sanmugamk  suggested, removed needs for extra column "reminder_last_sent" in user table.
So instead of keeping check on when was the last time we sent email to user,will set up cron weekly. May be on every Monday.
